base_layer/wallet/src/connectivity_service/handle.rs

```diff
@@ -61,11 +61,6 @@ impl WalletConnectivityHandle {
 #[async_trait::async_trait]
 impl WalletConnectivityInterface for WalletConnectivityHandle {
     fn set_base_node(&mut self, base_node_peer_manager: BaseNodePeerManager) {
-        if let Some(selected_peer) = self.base_node_watch.borrow().as_ref() {
-            if selected_peer.get_current_peer_id() == base_node_peer_manager.get_current_peer_id() {
-                return;
-            }
-        }
```

Comment:
Why remove this? Should we not return if the user selects the current peer again?

Reply:
You could also get a different set of peers on the list. It's a little hard to reason about this architecture. We can probably use a watch to notify when a peer ID changes and keep a list of appropriate peers in the connectivity service since that is where we decide to try another peer.

Reply:
But yes it is a minor issue that we reconnect if we select the same peer.